api/r/cellxgene.census/R/open.R

```diff
@@ -20,8 +21,28 @@
 #' census <- open_soma()
 #' as.data.frame(census$get("census_info")$get("summary")$read()$concat())
 #' census$close()
-open_soma <- function(census_version = "stable", uri = NULL, tiledbsoma_ctx = NULL) {
+open_soma <- function(
+    census_version = "stable",
+    uri = NULL,
+    tiledbsoma_ctx = NULL,
+    mirror = NULL) {
   if (is.null(uri) || is.null(tiledbsoma_ctx)) {
+    locator <- resolve_census_locator(census_version, uri, mirror)
+    uri <- locator$uri
+    if (is.null(tiledbsoma_ctx)) {
+      tiledbsoma_ctx <- new_SOMATileDBContext_for_census(NULL, mirror = locator$mirror_info)
+    }
+  }
+
+  return(tiledbsoma::SOMACollectionOpen(uri, tiledbsoma_ctx = tiledbsoma_ctx))
+}
+
+SUPPORTED_PROVIDERS <- c("S3", "file")
```

Comment:
We decided to add `unknown` for when the user opens the census with `uri=...`. This should be treated similarly to `file` (use a default context without adding S3 params like region).